### PR TITLE
Linux symlink support

### PR DESCRIPTION
**Pull request by Fernthedev**
Not tested, this is awaiting #14 too

### CODE REVIEW

File: QPM/Providers/RemoteQPMDependencyResolver.cs [hunk 1]
```diff
@@ -169,22 +169,35 @@ private void CopyAdditionalData(JsonElement elem, string root, string dst)
...
-                    var dest = Path.GetFullPath(Path.Combine(dst, item.GetString()));
-                    if (Directory.Exists(dest))
-                        Utils.DeleteDirectory(dest);
 
-                    // Create parent directories
-                    Utils.CreateDirectory(dst);
+                    var dest = Path.GetFullPath(Path.Combine(dst, item.GetString()!));
+                    // Add an extension so it "thinks" it's a file to get the parent directory easily
+                    var destFullParent =
+                        Path.GetDirectoryName(Path.TrimEndingDirectorySeparator(dest) + ".s")!;
```
Comment (sc2ad) [inline QPM/Providers/RemoteQPMDependencyResolver.cs:187]: Uhh, why?

Just do .TrimEnd(directory separator char) and call GetDirectoryName

File: QPM/SymLinker/LinkCreators/LinuxSymLinkCreator.cs [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
+using System.IO;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace SymLinker.LinkCreators
 {
     internal class LinuxSymLinkCreator : ISymLinkCreator
     {
-        internal LinuxSymLinkCreator() => throw new NotImplementedException("LinuxSymLinkCreator");
+        const string LIBC = "libc";
```
Comment (sc2ad) [inline QPM/SymLinker/LinkCreators/LinuxSymLinkCreator.cs:11]: Private

**Review by sc2ad — CHANGES_REQUESTED** [review]
No need to make a filename, can literally just trim end and call the function. Also make things that are const have proper access modifiers please.

Unrelated, I really should add a .editorconfig lol

File: QPM/Providers/RemoteQPMDependencyResolver.cs (hunk 1 above)
Comment (Fernthedev) [reply QPM/Providers/RemoteQPMDependencyResolver.cs:187]: Are you sure that won't just return the directory itself? I don't think it goes up one level, though I may be wrong

Comment (sc2ad) [reply QPM/Providers/RemoteQPMDependencyResolver.cs:187]: As long as it doesn't end with a `/`, it goes up a level. Here's a link if you don't believe me: [https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname?view=net-5.0](yoink)

Comment (Fernthedev) [reply QPM/Providers/RemoteQPMDependencyResolver.cs:187]: oh I'm stupid, alright I'll do that

**Review by sc2ad — APPROVED** [review]

**Pull request merged**